eigentrust-zk/src/verifier/transcript/mod.rs

```diff
@@ -132,14 +143,19 @@ where
 		self.state.update(&ec_point.inner.x.limbs);
 		self.state.update(&ec_point.inner.y.limbs);
 
+		self.native_state.update(&ec_point.inner.x.integer.limbs);
+		self.native_state.update(&ec_point.inner.y.integer.limbs);
```

Comment:
Why do we need native_state?

Reply:
The reason is that we need native scalar value for this op - `TranscriptReadChipset::squeeze_challenge`.
https://github.com/eigen-trust/protocol/pull/374/files#diff-11b415885c31eaa45e328fccdb132c3f01ff7015130d93ca21e7c7b929e03315R120-R136

I added the native scalar value as one field into the `Halo2LScalar` struct.
So, the above `squeeze_challenge` should have a way to get the native scalar value for `Halo2LScalar`.